### PR TITLE
Fix SDL audio glitching with steaming hot patches from Qrox

### LINKED ISSUE #63765: Police car's sound is distorted (@'s soundpack)
### Describe the bug

Sometimes CDDA plays strange clicking noises during its sound playback. Vormithrax encountered this problem a bunch of times on his streams. I can now repro this problem reliably in any police car when I drive.

### Attach save file

[Boulder Junction-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/10824351/Boulder.Junction-trimmed.tar.gz)

### Steps to reproduce

1. Enter any police car
2. Turn on the engine
3. Wait a couple of sound loops
(tried with @'s soundpack)

### Expected behavior

I expect the sound to not be distorted and to not produce clicks.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.1706 (21H2)
- Game Version: c69ea7c [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

This started happening about a month ago, so the problem has to be somewhat recent

### PR DESCRIPTION
**Pull request by akrieger**
#### Summary
None

#### Purpose of change
Fixes (for MSVC only) #63765. An SDL change last year (https://github.com/libsdl-org/SDL/commit/111c3add738587f97f41f8eaea35c6ee0d084ca5) introduced audio glitches/clicking when resampling audio files like in @'s soundpack. I identified https://github.com/libsdl-org/SDL/commit/1e5e8e2fda3796e76e6f7b1c39683925a3e9fed9 as a possible mitigation in https://github.com/libsdl-org/SDL_mixer/issues/466. @Qrox then wrote a better fix and put it up as a PR in https://github.com/libsdl-org/SDL/pull/7435. I tested CDDA building against SDL with that patch and it seems to work without introducing more regressions.

#### Describe the solution
I forked SDL's github repo and backported the fix commit to the 2.26.x release branch. I created a new tag based off the most recent one (created in the last day) which includes the fix. I then created a 'portfile overlay' inside the CDDA repo which contains a copy of the portfile from vcpkg, based off the vcpkg commit we are now using, except pointed at the patched release in my SDL fork. This then builds against the patched SDL.

#### Describe alternatives you've considered
Hosting a vcpkg fork to use as an alternate repository, and adding the portfile patch as a proper commit in that vcpkg repository. This would involve two forks to manage, and honestly the repositories feature is meant more for internal packages than patching upstream ones. 'Rebasing' the fix over future sdl fixes would be extremely marginally easier this way, but twice the forks is twice the pain, for just one package.

#### Testing
Downloaded @'s soundpack, loaded the save from #63765, reproduced the glitching sound. Build with my fork, did not reproduce the glitching sound.
```
1>Building sdl2[base,core]:x64-windows-static...
1>-- Installing port from location: C:\open\personal\Cataclysm-DDA\msvc-full-features\../.github/vcpkg_ports\sdl2
1>-- Downloading https://github.com/akrieger/SDL/archive/release-2.26.4_patched2.tar.gz -> akrieger-SDL-release-2.26.4_patched2.tar.gz...
1>-- Extracting source C:/open/personal/vcpkg/downloads/akrieger-SDL-release-2.26.4_patched2.tar.gz
```

### CODE REVIEW

File: .github/vcpkg_ports/sdl2/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,130 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO akrieger/SDL
+    REF release-2.26.4_patched2
+    SHA512 885243493d981eeeb87c116a9ad6c9ef66dd6f8fcc46023e6c539b72b8d454c051ce1a8d5a083b0494d888e0ea8a987ede7c0794f38bde8aaf8e7ae615740bdc
```
Comment (akrieger) [inline .github/vcpkg_ports/sdl2/portfile.cmake:5]: These are the only changed lines compared to clean vcpkg.

**Comment by Qrox** [comment]
> An SDL change last year introduced audio glitches/clicking in the msvc build

Is this bug present in the MinGW build? Asking because the MinGW CI build is recently updated to SDL 2.26.2.

**Comment by akrieger** [comment]
Probably. I'll try to track down exactly what I (believe) introduced it based on what seems to fix it.

**Comment by akrieger** [comment]
Yep, confirmed its in the mxe cross build.

**Comment by Qrox** [comment]
I'm trying a fix that uses integer arithmetics to fix this bug. I'll let you know if it works.

**Comment by akrieger** [comment]
In the meantime, should we take this anyway?

**Comment by Qrox** [comment]
I think my fix works. The clicking sound does not appear and a unit test shows that the signal-to-noise ratio of resampling increased from ~25 dB (using float or double) to ~50dB and the maximum error decreased from ~100% (using float) and ~5% (using double) to ~0.25%. I'll post a patch soon after I clean up the code.

**Pull request merged**